pkg/controller/linstorcsidriver/linstorcsidriver_controller.go

```diff
@@ -1054,17 +1054,29 @@ func defaultIfUnset(val, def int32) int32 {
 
 func getObjectMeta(controllerResource *piraeusv1.LinstorCSIDriver, nameFmt, component string) metav1.ObjectMeta {
 	return metav1.ObjectMeta{
-		Name:      fmt.Sprintf(nameFmt, controllerResource.Name),
-		Namespace: controllerResource.Namespace,
-		Labels: map[string]string{
-			"app.kubernetes.io/name":       kubeSpec.CSIDriverRole,
-			"app.kubernetes.io/instance":   controllerResource.Name,
-			"app.kubernetes.io/managed-by": kubeSpec.Name,
-			"app.kubernetes.io/component":  component,
-		},
+		Name:        fmt.Sprintf(nameFmt, controllerResource.Name),
+		Namespace:   controllerResource.Namespace,
+		Labels:      getCSILabels(controllerResource, component),
+		Annotations: controllerResource.Annotations,
 	}
 }
 
 func kubeletPath(csiResource *piraeusv1.LinstorCSIDriver, subdirs ...string) string {
 	return filepath.Join(append([]string{csiResource.Spec.KubeletPath}, subdirs...)...)
 }
+
+func getCSILabels(controllerResource *piraeusv1.LinstorCSIDriver, component string) map[string]string {
+	return mergeStringMap(map[string]string{
+		"app.kubernetes.io/name":       kubeSpec.CSIDriverRole,
+		"app.kubernetes.io/instance":   controllerResource.Name,
+		"app.kubernetes.io/managed-by": kubeSpec.Name,
+		"app.kubernetes.io/component":  component,
+	}, controllerResource.ObjectMeta.Labels)
```

Comment:
Same applies to this: we don't want the user controlled labels in the `Selector`